tests/shared/core/training_data/test_graph.py

```diff
@@ -45,3 +47,24 @@ def test_node_ordering_with_cycle():
 
 def test_is_empty():
     assert StoryGraph([]).is_empty()
+
+
+async def test_consistent_fingerprints():
```

Comment:
Do we have any tests that test the uniqueness of the generated checkpoint names?